rustls/src/webpki/client_verifier.rs

```diff
@@ -42,6 +44,35 @@ impl ClientCertVerifierBuilder {
         }
     }
 
+    /// Replace the list of trust anchor subjects sent to the client as hints of acceptable trust
+    /// anchors during the TLS handshake.
+    ///
+    /// By default, the client cert verifier will [`RootCertStore::subjects`] with the root cert
+    /// store configured for client certificate authentication for the hint subjects.
+    ///
+    /// See [`ClientCertVerifier::client_auth_hint_root_subjects`] for more information on
+    /// circumstances where you may want to override the default hint subjects.
+    pub fn with_hint_subjects(
+        mut self,
+        subjects: impl IntoIterator<Item = DistinguishedName>,
+    ) -> Self {
+        self.hint_subjects = subjects.into_iter().collect();
+        self
+    }
+
+    /// Add an additional [`DistinguishedName`] to the list of trust anchor subjects sent to the
+    /// client as hints of acceptable trust anchors during the TLS handshake.
+    ///
+    /// By default, the client cert verifier will [`RootCertStore::subjects`] with the root cert
+    /// store configured for client certificate authentication for the hint subjects.
+    ///
+    /// See [`ClientCertVerifier::client_auth_hint_root_subjects`] for more information on
+    /// circumstances where you may want to override the default hint subjects.
+    pub fn add_hint_subject(mut self, subject: DistinguishedName) -> Self {
```

Comment:
Given that a one-element array implements IntoIter, not sure this method pulls its weight?

Reply:
> Given that a one-element array implements IntoIter, not sure this method pulls its weight?

True, but `add_hint_subject` _adds_ to the existing `Vec` whereas the other method replaces it outright.

I think there's value in both: if we only have this method then it isn't possible to replace the defaults, or specify an empty list to send. If we only have the other method then you have to clone the `RootCertStore` from the one used for authentication, add your extra subject to it, and then use the other fn to replace the whole list.

For a use-case like the one described in #1549 where you're happy with the default but just want to add one or two other subjects from intermediate certs it seemed helpful to have this fn.

Reply:
Maybe there's an argument to be made that this should be `add_hint_subjects(mut self, subjects: impl IntoIterator<Item = DistinguishedName>) ` for consistency and to make it easier to use with multiple names?

Reply:
Ah, sorry, missed the add vs replace difference. I guess it comes down to whether there's a use case for removing some of the hints that refer back to the set anchors...

Reply:
> I guess it comes down to whether there's a use case for removing some of the hints that refer back to the set anchors...

I'm not sure about removing one or two, but replacing the whole list with the empty set should be supported based on the text Jsha cited in #1549:
>  To mitigate this failure, the server shall either: 1) maintain the trust anchors of the various PKIs whose subscribers are the potential clients for the server and include them in the hints list or 2) be configured to send an empty hints list so that the client can always provide a certificate it possesses.

Reply:
Okay, in that case I think your method set probably makes sense.

Reply:
> Maybe there's an argument to be made that this should be add_hint_subjects(mut self, subjects: impl IntoIterator<Item = DistinguishedName>)  for consistency and to make it easier to use with multiple names?

I ended up making this change. If you only have one `DistinguishedName` it's easy enough to make a slice to give, and in other cases the symmetry is helpful. 

> I wonder if we should use a different name for the replacing method? The with_() doesn't feel very explicit about appending vs replacing.

I left the name as-is (missed your edit), but I could change it to `replace_root_hint_subjects` if you think that would be more explicit.

Reply:
i think realistically either people will want to add to the existing hints, or have them empty -- i can't see why people would want to send _some_ hints, but not the ones for actual issuer names they trust.

how about `clear_root_hint_subjects(mut self)` (a good venue for recording that this means "clients will make a free choice of certificate to send" in a comment) and `add_root_hint_subjects(mut self, subjects: impl IntoIterator<Item = DistinguishedName>)`?

of course, those can be combined to set whatever contents someone likes.

Reply:
Yeah, that's probably the more explicit option.

Reply:
SGTM

Reply:
Updated to replace w/ `clear_root_hint_subjects`.